### PR TITLE
upgrades dependencies

### PR DESCRIPTION
**Pull request by bockstaller**
Upgrades all dependencies to the newest state that doesn't require large changes.

The largest change is the removal of `await client.onReady();` caused by  https://github.com/microsoft/vscode-languageserver-node/blob/main/README.md#3170-protocol-800-json-rpc-800-client-and-800-server

After this PR, we could move to eslint (and prettier?) as tslint is deprecated.

### CODE REVIEW

**Comment by chadlwilson** [comment]
You’ll need to rebase off master as I fixed the build the other day to ensure a minimalist change was made that had all the functional tests passing before anything further was done.

**Comment by bockstaller** [comment]
@chadlwilson done

**Comment by chadlwilson** [comment]
Thanks for this! I did briefly try the vscode dep updates but didn't look any deeper as to why it wasn't working :-D

Have you given it a test drive "in the real world"? :-)

**Comment by bockstaller** [comment]
Yep, I am running it locally and I've seen no anomalies so far

**Review by chadlwilson — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -461,30 +461,30 @@
...
+    "clipboardy": "^4.0.0",
     "fs-extra": "^11.1.1",
     "get-port": "^5.1.1",
     "ps-tree": "^1.2.0",
-    "vscode-jsonrpc": "^6.0.0",
-    "vscode-languageclient": "^7.0.0",
-    "vscode-languageserver-protocol": "~3.16.0",
+    "vscode-jsonrpc": "^8.2.0",
+    "vscode-languageclient": "^9.0.1",
+    "vscode-languageserver-protocol": "^3.17.5",
```
[the remark this answers is not in the document]
Comment (bockstaller) [reply package.json]: Done

File: package.json [hunk 2]
```diff
@@ -461,30 +461,30 @@
...
-    "terser-webpack-plugin": "^5.3.0",
-    "ts-loader": "^9.2.6",
+    "terser-webpack-plugin": "^5.3.9",
+    "ts-loader": "9.5.0",
     "ts-mockito": "^2.6.1",
     "tslint": "^6.1.3",
     "typescript": "^5.2.2",
     "vsce": "^2.15.0",
-    "webpack": "^5.89.0",
+    "webpack": "5.89.0",
```
[the remark this answers is not in the document]
Comment (bockstaller) [reply package.json]: Done

File: package.json [hunk 3]
```diff
@@ -461,30 +461,30 @@
...
+    "@types/vscode": "^1.83.1",
+    "@vscode/test-electron": "^2.3.6",
     "cross-env": "^7.0.3",
     "glob": "^10.3.10",
     "mocha": "^10.2.0",
     "sinon": "^17.0.0",
-    "terser-webpack-plugin": "^5.3.0",
-    "ts-loader": "^9.2.6",
+    "terser-webpack-plugin": "^5.3.9",
+    "ts-loader": "9.5.0",
```
[the remark this answers is not in the document]
Comment (bockstaller) [reply package.json]: Done

**Review by chadlwilson — APPROVED** [review]

**Pull request merged**